autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
+            current_hash, __ = current_checksum.split()
+            host = self.config.get('ncbi', 'host')
+            remote_checksum = self.get_remote_checksum(host, 'nr')
+            remote_hash, __ = remote_checksum.split()
+            if current_hash == remote_hash:
+                logger.debug(f'nr checksums match, skipping...')
+                checksums_match = True
+
+        no_update_db = os.path.exists(db_outfpath) and not self.do_update
+        if self.dryrun or no_update_db or checksums_match:
+            self.config.set('ncbi', 'nr', db_outfpath)
+            logger.debug(f'set ncbi nr: {db_outfpath}')
+            return
+
+        diamond.makedatabase(fasta=db_infpath, database=db_outfpath, nproc=self.nproc)
+        if os.path.basename(db_infpath) == 'nr.gz':
+            # nr.gz will be removed after successful nr.dmnd construction
+            write_checksum(db_infpath, db_infpath_md5)
+            os.remove(db_infpath)
+
         self.config.set('ncbi', 'nr', db_outfpath)
         logger.debug(f'set ncbi nr: {db_outfpath}')
 
     def extract_taxdump(self):
-        """Extract autometa required files from ncbi taxdump directory.
+        """Extract autometa required files from ncbi taxdump.tar.gz archive
+        into ncbi databases directory and update user config with extracted paths.
+
+        This only extracts nodes.dmp, names.dmp and merged.dmp from
+        taxdump.tar.gz if the files do not already exist. If `do_update`
+        was originally supplied as `True` to the Databases instance, then the
+        previous files will be replaced by the new taxdump files.
 
-        Extracts nodes.dmp, names.dmp and merged.dmp from taxdump.tar.gz
+        After successful extraction of the files, a checksum will be written of the
+        archive for future checking and then the archive will be removed to save user disk space.
 
         Returns
         -------
         NoneType
+            Will update `self.config` section `ncbi` with options 'nodes','names','merged'
 
         """
-        taxdump = self.config.get('ncbi', 'taxdump')
+        taxdump_fpath = self.config.get('ncbi','taxdump')
         taxdump_files = [
             ('nodes', 'nodes.dmp'),
             ('names', 'names.dmp'),
             ('merged', 'merged.dmp'),
         ]
-        for option, fname in taxdump_files:
+        for option,fname in taxdump_files:
             outfpath = os.path.join(self.ncbi_dir, fname)
-            if not self.dryrun and not os.path.exists(outfpath):
-                outfpath = untar(taxdump, self.ncbi_dir, fname)
+            if self.dryrun:
+                logger.debug(f'UPDATE (ncbi,{option}): {outfpath}')
+                self.config.set('ncbi', option, outfpath)
+                continue
+            # Only update the taxdump files if the user says to do an update.
+            if self.do_update and os.path.exists(outfpath):
+                os.remove(outfpath)
```

Comment:
Here, wouldn't we need to also check that the `taxdump` file exists, because previous runs of this function would have deleted it? The below code assumes it exists.